File: leetcode/rotate_array.py
class Solution:
    def rotate(self, nums: List[int], k: int) -> None:
        if len(nums) == 0 or k == 0 : return nums
        n = len(nums)
        k %= n
        nums[:] = nums[n-k:] + nums[:n-k]
        return nums
# class Solution:
#   def rotate(self, nums: List[int], k: int) -> None:
#       n = len(nums)
#       k %= n

#       for i in range(0, n-k):
#           nums.append(nums[0])
#           del nums[0]
#       return nums

File: leetcode/rotate_array.cpp
class Solution {
public:
    void rotate(vector<int>& nums, int k) {
        int size = nums.size();
        k %= size;
        if (k != 0){
            k = size-k;
            vector<int> FirstkElts;
            for (int i = 0; i < k; i++) {
                FirstkElts.push_back(nums[i]);
            }
            nums.erase(nums.begin(), nums.begin()+k);
            for (int i = 0; i < k; i++){
                nums.push_back(FirstkElts[i]);
            }
        }
    }
};
